### PR TITLE
Replace pub dim field of writers with constructor

### PR DESCRIPTION
**Pull request by pka**
#156

### CODE REVIEW

**Review by nyurik — COMMENTED** [review]
a few minor nits

File: geozero/src/geojson/geojson_writer.rs [hunk 1]
```diff
@@ -5,17 +5,20 @@ use std::io::Write;
...
 pub struct GeoJsonWriter<W: Write> {
-    pub dims: CoordDimensions,
+    dims: CoordDimensions,
     out: W,
 }
 
 impl<W: Write> GeoJsonWriter<W> {
     pub fn new(out: W) -> Self {
-        Self {
+        GeoJsonWriter {
```
[the remark this answers is not in the document]
Comment (pka) [reply geozero/src/geojson/geojson_writer.rs:14]: Seems like we have a different taste for once...
I prefer concrete struct names for readability here.

**Review by nyurik — APPROVED** [review]
lol, oki, as long as its not `Default::default` :smile:

File: geozero/src/gdal/gdal_writer.rs [hunk 2]
```diff
@@ -17,6 +17,12 @@ impl GdalWriter {
     pub fn new() -> Self {
         Self::default()
     }
+    pub fn with_dims(dims: CoordDimensions) -> Self {
+        GdalWriter {
+            dims,
+            ..Default::default()
```
[the remark this answers is not in the document]
Comment (michaelkirk) [reply geozero/src/gdal/gdal_writer.rs]: Wait, why is `Default::default()` considered evil?

[the remark this answers is not in the document]
Comment (nyurik) [reply geozero/src/gdal/gdal_writer.rs]: because it is not as readable as a concrete type or `Self` :)

**Pull request merged**